src/Collections/Artemis.Plugins.Audio/DataModelExpansion/AudioEndpointVolumeModule.cs

```diff
@@ -0,0 +1,228 @@
+
+            lock (_audioEventLock)
+            {
+                for (int i = 0; i < _channelsDataModels.Count; i++)
+                {
+                    DynamicChild<AudioChannelDataModel> audioChannelDataModel = _channelsDataModels[i];
+                    float volumeNormalized = _audioEndpointVolume.Channels[i].VolumeLevelScalar;
+                    audioChannelDataModel.Value.VolumeNormalized = volumeNormalized;
+                    audioChannelDataModel.Value.Volume = volumeNormalized * 100f;
+                }
+            }
+        }
+
+        private void PopulateChannels()
+        {
+            DataModel.Channels.ClearDynamicChildren();
+            _logger.Verbose($"AudioEndpoint device {_audioDevice.FriendlyName} channel list cleared");
+            _logger.Verbose($"Preparing to populate {_audioEndpointVolume.Channels.Count} channels for device {_audioDevice.FriendlyName}");
+            _channelsDataModels.Clear();
+            for (int i = 0; i < _audioEndpointVolume.Channels.Count; i++)
+            {
+                _channelsDataModels.Add(
+                    DataModel.Channels.AddDynamicChild(i.ToString(), new AudioChannelDataModel { ChannelIndex = i }, $"Channel {i}")
+                );
+                _logger.Verbose($"AudioEndpoint device {_audioDevice.FriendlyName} channel {i} populated");
+            }
+        }
+
+        #endregion
+
+        #region Audio Management methods
+
+        private void NotificationClient_DefaultDeviceChanged()
+        {
+            _audioDeviceChanged = true;
+            // Workarround. MMDevice won't dispose if Dispose() is called from 
+            // non parent thread and NaudioNotificationClient callbacks come from another thread.
+            // We will use Update() mrhod to dispose MMDevice from creator thread because this (NotificationClient_DefaultDeviceChanged()) is called from another thread
+        }
+
+        private void UpdateAudioEndpointDevice(bool firstRun = false)
+        {
+            lock (_audioEventLock)
+            {
+                if (!firstRun) FreeAudioEndpointDevice();
+
+                if (SetAudioEndpointDevice())
+                {
+                    PopulateChannels();
+                    _audioDeviceChanged = false;
+                    UpdateVolumeDataModel();
+                }
+            }
+        }
+
+        private void FreeAudioEndpointDevice()
+        {
+            string disposingAudioEndpointDeviceFriendlyName = _audioDevice?.FriendlyName ?? "Unknown";
+            _audioDevice?.Dispose();
+            _audioDevice = null;
+            _logger.Verbose($"AudioEndpoint device {disposingAudioEndpointDeviceFriendlyName} unregistered as source device to fill AudioEndpoint volume data model");
+            DataModel.Reset();
+        }
+
+        private bool SetAudioEndpointDevice()
+        {
+            _audioDevice = _naudioDeviceEnumerationService.GetDefaultAudioEndpoint(DataFlow.Render, Role.Console);
```

Comment:
```cs
_audioDevice = _naudioDeviceEnumerationService.GetDefaultAudioEndpoint(_flow, _role);
```

Reply:
This way at least the module will correctly capture the mic. Unfortunately, the volume doesn't seem to update unless i also have the native windows mixer open:

https://github.com/Artemis-RGB/Artemis.Plugins/assets/29486064/f4683ebb-1c20-435c-b9c5-eb2c61658c43

I'm not sure if we need to listen for some event or call some update function.

Reply:
Whoops, missed that one (\_flow/\_role). That's a major issue. Fix implemented.

In regards to the volume not updating, I can confirm this issue occurs on my system. My suspicion is that nAudio is registering the device and pulling in data - but not actually opening access to the device itself, in regards to the security context. If you note the icon in the system tray for "Apps using your microphone", it will appear when the Windows recording device panel is enabled, and disappear when it is not. The same occurs if you use the Settings sound panel (the "replacement" control panel), or (for another example) if you have a Discord call open. I'll do a little more digging.

Reply:
OK, we do have to initialise a recording access in order for the volume data to be updated. This was outside my use case, as I only wanted the mute state or otherwise; but better to do it right the first time.

Reply:
In my experience users don't like it when apps listen to the mic 24/7 so I would be careful with how you implement this. Maybe separate volume and actual sounds into separate modules or add an option to enable mic capture or something? It's not a trivial feature to add imo.